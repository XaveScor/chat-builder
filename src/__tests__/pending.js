import * as React from 'react';
import {createPage, ConformsForm, createPending, input, Stop, questionBubble, useChatBuilder} from '..'
import * as renderer from 'react-test-renderer'

async function delay(timeout) {
    return new Promise(resolve => {
      setTimeout(resolve, timeout);
    });
  }

const pending = createPending({
    component: () => 'pending',
})
const pendingConfig = {
    pending,
    input,
}

it('ConformsForm', done => {
    const startPage = createPage(async () => {
        await delay(700)
        return {
            steps: [
                {
                    question: questionBubble,
                    questionProps: {
                        question: 'question',
                    },
                    input,
                },
            ],
            nextPage: Stop,
        }
    })

    let tree
    renderer.act(() => {
        tree = renderer
            .create(<ConformsForm page={startPage} pending={pendingConfig} />)
    })
    
    setTimeout(() => {
        const res = tree.toJSON()
        expect(res).toMatchSnapshot()
        setTimeout(() => {
            const res = tree.toJSON()
            expect(res).toMatchSnapshot()
            done()
        }, 700)
    }, 600)
});

it('useChatBuilder', done => {
    const startPage = createPage(async () => {
        await delay(700)
        return {
            steps: [
                {
                    question: questionBubble,
                    questionProps: {
                        question: 'question',
                    },
                    input,
                },
            ],
            nextPage: Stop,
        }
    })

    const Form = (props) => {
        const Chat = useChatBuilder(startPage, {
            pending: pendingConfig,
        })

        return <Chat {...props} />
    }

    let tree
    renderer.act(() => {
        tree = renderer.create(<Form />)
    })
    
    setTimeout(() => {
        const res = tree.toJSON()
        expect(res).toMatchSnapshot()
        setTimeout(() => {
            const res = tree.toJSON()
            expect(res).toMatchSnapshot()
            done()
        }, 700)
    }, 600)
});
