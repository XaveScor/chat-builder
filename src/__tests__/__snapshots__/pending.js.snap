// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConformsForm 1`] = `
Array [
  "pending",
  <input
    onChange={[Function]}
  />,
  <input
    type="submit"
  />,
]
`;

exports[`ConformsForm 2`] = `
<p
  className="robot"
>
  question
</p>
`;

exports[`useChatBuilder 1`] = `
Array [
  "pending",
  <input
    onChange={[Function]}
  />,
  <input
    type="submit"
  />,
]
`;

exports[`useChatBuilder 2`] = `
<p
  className="robot"
>
  question
</p>
`;
